src/public/events/events.service.ts

```diff
@@ -30,89 +30,150 @@ export class EventsService {
     });
   }
   async update(appId: string, eventId: string, updateEventDto: UpdateEventDto) {
-    if (updateEventDto.name) {
-      const slug = slugify(updateEventDto.name, {
+    const { name, eventLocation, ...eventData } = updateEventDto;
+
+    const existingEvent = await this.database.event.findUnique({
+      where: {
+        id: eventId,
+        appId
+      },
+      include: {
+        EventLocation: true
+      }
+    });
+
+    if (!existingEvent) {
+      throw new NotFoundException("Event not found");
+    }
+
+    const updateData: Partial<UpdateEventDto> = {
+      ...eventData
+    };
+
+    if (name && name !== existingEvent.name) {
+      const slug = slugify(name, {
         lower: true,
         strict: true,
         trim: true
       });
-      updateEventDto["slug"] = slug;
-    }
 
-    const isNameExists = await this.database.event.findFirst({
-      where: {
-        slug: updateEventDto.slug,
-        appId,
-        NOT: {
-          id: eventId
+      const isSlugTaken = await this.database.event.findFirst({
+        where: {
+          slug,
+          appId,
+          NOT: {
+            id: eventId
+          }
         }
+      });
+
+      if (isSlugTaken) {
+        throw new ConflictException("Event with this name already exists");
```

Comment:
i think it would be nice if we have interpolated the passed name of the event in the Error (we have it in other places so it woudl feel more consisnent)